UnityProject/Assets/Scripts/Effects/CustomEffectBehaviour.cs

```diff
@@ -2,7 +2,7 @@
 
 public class CustomEffectBehaviour : MonoBehaviour
 {
-	public ParticleSystem particleSystem;
+	public new ParticleSystem particleSystem;
```

Comment:
why new?

Reply:
MonoBehaviour already has a `particleSystem` member. So we just ignore it and use our (assumed serialized) one instead. I could just rename the variable.

Reply:
Renaming is probably the best solution